internal/repository/validator.go

```diff
@@ -0,0 +1,180 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+package repository
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/cartesi/rollups-node/internal/node/model"
+	"github.com/jackc/pgx/v5"
+)
+
+const DefaultServiceTimeout = 5 * time.Minute
+
+func (pg *database) GetLastProcessedBlock(
+	ctx context.Context,
+	appAddress Address,
+) (uint64, error) {
+	var block uint64
+
+	query := `
+	SELECT
+		last_processed_block
+	FROM
+		application
+	WHERE
+		address=@address`
+
+	args := pgx.NamedArgs{
+		"address": appAddress,
+	}
+
+	err := pg.db.QueryRow(ctx, query, args).Scan(&block)
+	if err != nil {
+		return 0, fmt.Errorf("QueryRow failed: %v\n", err)
+	}
+
+	return block, nil
+}
+
+func (pg *database) GetAllOutputsFromProcessedInputs(
+	ctx context.Context,
+	startBlock uint64,
+	endBlock uint64,
+	timeout *time.Duration,
+) ([]Output, error) {
+	query := `
+	SELECT
+		o.id,
+		o.index,
+		o.raw_data,
+		o.input_id,
+		i.status
+	FROM
+		output o
+	INNER JOIN
+		input i
+	ON
+		o.input_id=i.id
+	WHERE
+		i.block_number BETWEEN @startBlock and @endBlock
+	ORDER BY
+		o.index asc`
+
+	args := pgx.NamedArgs{
+		"startBlock": startBlock,
+		"endBlock":   endBlock,
+	}
+
+	var to *time.Timer
```

Comment:
This timeout logic should be separate from the SQL logic.
The function with the timeout logic should call the SQL logic function.
We can also use `context.WithDeadline`.